### PR TITLE
Enforce Password Length

### LINKED ISSUE #14: Password Length in Client.py
Force user to choose an apropriate password length when they register. 12 - 15 chars long. Add a unit_test

### PR DESCRIPTION
**Pull request by jjjoycing123**
### Changes
* Adds password length check
* Adds unit test
* Closes #14

### CODE REVIEW

File: securedrop/client.py [hunk 1]
```diff
@@ -90,7 +90,12 @@ def register_new_user(self):
 
         pw1 = getpass.getpass(prompt="Enter Password: ")
         pw2 = getpass.getpass(prompt="Re-enter password: ")
+
         if name and email and pw1 and pw2:
+            #enforce password length to min of 12 characters
```
Comment (aokellermann) [inline securedrop/client.py]: Put space between `#` and comment

File: securedrop/client.py [hunk 2]
```diff
@@ -90,7 +90,12 @@ def register_new_user(self):
...
+            if (len(pw1) < 12) or (len(pw2) < 12):
```
Comment (aokellermann) [inline securedrop/client.py]: You can move this below the auth equality check so only one input's length needs to be checked.

File: securedrop/client.py [hunk 3]
```diff
@@ -90,7 +90,12 @@ def register_new_user(self):
...
+                raise RuntimeError("Invalid input")
```
Comment (aokellermann) [inline securedrop/client.py]: Error message should be more descriptive

File: securedrop/client.py [hunk 4]
```diff
@@ -105,7 +110,6 @@ def register_new_user(self):
         else:
             raise RuntimeError("Invalid input")
 
-
```
Comment (aokellermann) [inline securedrop/client.py]: Remove this newline

File: securedrop/tests/client_test.py [hunk 5]
```diff
@@ -81,64 +90,65 @@ def assert_initial_registered_users_is_valid(self, ru):
             self.assertTrue(cd.auth.salt)
             self.assertTrue(cd.auth.key)
 
-    def test_aae_initial_json_valid(self):
+
```
Comment (aokellermann) [inline securedrop/tests/client_test.py]: Remove this newline

**Review by aokellermann — CHANGES_REQUESTED** [review]

File: securedrop/client.py [hunk 6]
```diff
@@ -105,7 +110,6 @@ def register_new_user(self):
         else:
             raise RuntimeError("Invalid input")
```
Comment (BobbyBytes) [inline securedrop/client.py]: Add more specific error message to differ "between password too short", and "passwords different".

File: securedrop/tests/client_test.py (hunk 5 above)
Comment (BobbyBytes) [inline securedrop/tests/client_test.py]: Remove extra line

**Review by BobbyBytes — COMMENTED** [review]
Looks good, 
Just a couple of comments I had

File: securedrop/client.py (hunk 4 above)
Comment (jjjoycing123) [reply securedrop/client.py]: Gonna let the formater take care of this.

File: securedrop/tests/client_test.py (hunk 5 above)
Comment (jjjoycing123) [reply securedrop/tests/client_test.py]: Gonna let the formater take care of this.

File: securedrop/client.py (hunk 3 above)
Comment (jjjoycing123) [reply securedrop/client.py]: Changed to: "Password is too short! Password must be at least 12 characters"

File: securedrop/client.py (hunk 6 above)
Comment (jjjoycing123) [reply securedrop/client.py]: Changed to: "Password is too short! Password must be at least 12 characters"

File: securedrop/tests/client_test.py (hunk 5 above)
Comment (jjjoycing123) [reply securedrop/tests/client_test.py]: Gonna let the formater take care of this.

File: securedrop/client.py (hunk 1 above)
Comment (aokellermann) [reply securedrop/client.py]: Not resolved

File: securedrop/client.py (hunk 2 above)
Comment (aokellermann) [reply securedrop/client.py]: This can be removed now

**Review by aokellermann — CHANGES_REQUESTED** [review]

File: securedrop/client.py (hunk 1 above)
Comment (jjjoycing123) [reply securedrop/client.py]: Should be resolved now

File: securedrop/tests/client_test.py [hunk 7]
```diff
@@ -57,10 +57,19 @@ def test_aac_initial_ask_to_register_empty_input(self):
                         with self.assertRaises(RuntimeError):
                             Client(sd_filename).login()
 
+    def test_initial_ask_to_register_password_too_short(self):
```
Comment (aokellermann) [inline securedrop/tests/client_test.py]: Test function name should follow similar convention to others with prefix being `test_aai` where `i` is the `ith` test to run. I couldn't find a better way to ensure deterministic test order -- I know it's clunky. This is the perfect spot for the function, so it should be `test_aad`. The tests below will need to have their prefixes adjusted accordingly to maintain the same order after the insertion, so replace the previous `test_aad`with `test_aae` and so on.

**Review by aokellermann — CHANGES_REQUESTED** [review]

File: securedrop/tests/client_test.py (hunk 7 above)
Comment (jjjoycing123) [reply securedrop/tests/client_test.py]: Will do.

Comment (jjjoycing123) [reply securedrop/tests/client_test.py]: done

**Review by aokellermann — APPROVED** [review]
Nice work!

**Pull request merged**